ts/weekView.ts

```diff
@@ -0,0 +1,520 @@
+	container.className = "weekView-column hours";
+
+	const headerCell = document.createElement("div");
+	headerCell.className = "hour header-cell";
+
+	const hourSeparator = document.createElement("div");
+	hourSeparator.className = "hour-separator";
+
+	headerCell.appendChild(hourSeparator);
+
+	container.appendChild(headerCell);
+
+	for (let index = 1; index < HOUR_COUNT; index++) {
+		const cell = createHourCell(
+			new Date(0, 0, 0, index).valueOf(),
+			dateFormatter
+		);
+		container.appendChild(cell);
+	}
+
+	return container;
+};
+
+const createWeekView = (
+	date: Date,
+	dateFormatter: DateFormatter
+): HTMLElement => {
+	const weekView = document.createElement("div");
+	weekView.className = "weekView-main";
+
+	weekView.appendChild(createHoursColumn(dateFormatter));
+
+	weekView.addEventListener("click", (e) => {
+		const eventTarget = e.target as HTMLElement;
+		const eventTargetDataset = eventTarget.dataset;
+		if (eventTargetDataset.timestamp) {
+			showFormModal(new Date(parseInt(eventTargetDataset.timestamp)));
+		}
+	});
+
+	const weekDates = getWeekDates(date);
+
+	for (const date of weekDates) {
+		const column = createDayColumn(date, dateFormatter);
+		weekView.appendChild(column);
+	}
+
+	return weekView;
+};
+
+let prevWeekView: HTMLElement | null;
+let prevTimeout: number;
+
+export const switchWeekView = (
+	date: Date,
+	prevDate: Date,
+	dateFormatter: DateFormatter
+): void => {
+	if (prevWeekView) {
+		clearTimeout(prevTimeout);
+		prevWeekView?.remove();
+		prevWeekView = null;
+	}
+
+	const wrapper = document.querySelector(".weekView-wrapper") as HTMLElement;
+	const weekView_current = document.querySelector(
+		".weekView-main"
+	) as HTMLElement;
+
+	const weekView_new = createWeekView(date, dateFormatter);
+	wrapper.appendChild(weekView_new);
+
+	if (prevDate && !isSameWeek(date, prevDate)) {
+		const slideInClass = date > prevDate ? "slideIn_ltr" : "slideIn_rtl";
+		weekView_new.classList.add(slideInClass);
+	}
+
+	prevWeekView = weekView_current;
+	prevTimeout = setTimeout(() => {
+		weekView_current?.remove();
+		weekView_new.classList.remove("slideIn_ltr");
+		weekView_new.classList.remove("slideIn_rtl");
+	}, 200);
+};
+
+export const init = (dateFormatter: DateFormatter): void => {
+	const modalContainer = document.querySelector("#eventCardModal");
+
+	if (modalContainer) {
+		modalContainer.addEventListener("click", (e) => {
+			if ((e.target as HTMLElement).id === "eventCardModal") {
+				hideModal();
+			}
+		});
+	}
+
+	switchWeekView(selectedDate.value, selectedDate.prev, dateFormatter);
+	updateDayLabels(selectedDate.value, dateFormatter);
+
+	localeState.addListener(() => {
+		const weekViewMain = document.querySelector(".weekView-main");
+		weekViewMain!.innerHTML = "";
+		weekViewMain?.appendChild(
+			createWeekView(selectedDate.value, dateFormatter)
+		);
+	});
+	selectedDate.addListener((currentState: Date, prevState: Date) =>
+		switchWeekView(currentState, prevState, dateFormatter)
+	);
+};
```

Comment:
This file is 520 lines long. Maybe there is functionality here that isn't responsible for `weekView` and could be separated ? I'm not sure, just asking.